CalibTracker/SiStripChannelGain/plugins/SiStripGainFromCalibTree.cc

```diff
@@ -178,16 +193,20 @@ class SiStripGainFromCalibTree : public ConditionDBWriter<SiStripApvGain> {
 	std::string  OutputGains;
 	vector<string> VInputFiles;
 
-	MonitorElement*        Charge_Vs_Index;
-	MonitorElement*        Charge_Vs_Index_Absolute;
-	MonitorElement*        Charge_Vs_PathlengthTIB;
-	MonitorElement*        Charge_Vs_PathlengthTOB;
-	MonitorElement*        Charge_Vs_PathlengthTIDP;
-	MonitorElement*        Charge_Vs_PathlengthTIDM;
-	MonitorElement*        Charge_Vs_PathlengthTECP1;
-	MonitorElement*        Charge_Vs_PathlengthTECP2;
-	MonitorElement*        Charge_Vs_PathlengthTECM1;
-	MonitorElement*        Charge_Vs_PathlengthTECM2;
+        enum statistic_type {None=-1, StdBunch, StdBunch0T, FaABunch, FaABunch0T, IsoBunch, IsoBunch0T, Harvest};
+
+        std::vector<string>    dqm_tag_;
```

Comment:
@davidlange6

yes they could but what is the advantage of translating them into arrays? Please explain.

Unless there is a strong motivation I would rather prefer to have them as std::vector because this implementation is more robust / modular with respect further addition of items. If you are concerned about  dynamic memory allocation or fragmentation,  I can add a reserve statement in the tool constructor in order to pre-allocate the memory I need. Please let me know whether this is another possible solution.

By the way: my UID is currently being reassigned and , so, I cannot login anymore on lxplus and watch to my file on AFS. Hope this is a temporary problem and tomorrow the situation is back to normal. I say thins because I amy not be very responsive on implementing such change.

 


> On 22 Mar 2016, at 19:39, David Lange <notifications@github.com> wrote:
> 
> In CalibTracker/SiStripChannelGain/plugins/SiStripGainFromCalibTree.cc <https://github.com/cms-sw/cmssw/pull/13724#discussion_r57044515>:
> 
> > @@ -178,16 +193,20 @@ class SiStripGainFromCalibTree : public ConditionDBWriter<SiStripApvGain> {
> >  	std::string  OutputGains;
> >  	vector<string> VInputFiles;
> >  
> > -	MonitorElement*        Charge_Vs_Index;
> > -	MonitorElement*        Charge_Vs_Index_Absolute;
> > -	MonitorElement*        Charge_Vs_PathlengthTIB;
> > -	MonitorElement*        Charge_Vs_PathlengthTOB;
> > -	MonitorElement*        Charge_Vs_PathlengthTIDP;
> > -	MonitorElement*        Charge_Vs_PathlengthTIDM;
> > -	MonitorElement*        Charge_Vs_PathlengthTECP1;
> > -	MonitorElement*        Charge_Vs_PathlengthTECP2;
> > -	MonitorElement*        Charge_Vs_PathlengthTECM1;
> > -	MonitorElement*        Charge_Vs_PathlengthTECM2;
> > +        enum statistic_type {None=-1, StdBunch, StdBunch0T, FaABunch, FaABunch0T, IsoBunch, IsoBunch0T, Harvest};
> > +
> > +        std::vector<string>    dqm_tag_;
> could these vectors all be std::arrays? They look to be a hardcoded length
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub <https://github.com/cms-sw/cmssw/pull/13724/files/588628c5918e3dc4da22ac8b6a0533d05b2fb855#r57044515>